wordpress-mobile/WordPress-Android
# Api26/jobschedule reader update service

This PR implements a distinction on pre and post API 26 and makes use of the standard Service or a `JobService` starting from API 26, for the `ReaderUpdateService` as explained in https://github.com/wordpress-mobile/WordPress-Android/pull/7506/#issuecomment-378949294.

In order to do that, the following steps were made:
- extracted the business logic from `ReaderUpdateService` out to a shared class named `ReaderUpdateLogic`.
- Made a specific implementation of the `JobService` for API 26+, thus effectively having differentiated Services
- Declared such a new JobService in the manifest
- a new interface, called `ServiceCompletionListener` has been introduced so both implementations can keep up with finishing worker threads / scheduled jobs
- finally, a helper class named `ReaderUpdateServiceStarter`was introduced for consumers to be able to seamlessly call `startService`and let it handle which implementation to use, depending on the platform we're running on.

This pattern will be followed in other Services in the list in https://github.com/wordpress-mobile/WordPress-Android/pull/7506/#issuecomment-378949294, and we'll use the very handy `JobIntentService` for any other remaining `IntentServices` out there (specifically from that list, `PublicizeUpdateService`).

To test:
- on API26, run the app and place breakpoints in `ReaderUpdateJobService` to make sure it's executed.
- On previous Android versions, run the app and place breakpoints in `ReaderUpdateService` to make sure it's executed

cc @nbradbury
